packages/hierarchies/src/hierarchies/internal/TreeNodesReader.ts

```diff
@@ -28,8 +31,12 @@ export class TreeQueryResultsReader {
   }
 
   public async *read(queryExecutor: ILimitingECSqlQueryExecutor, query: ECSqlQueryDef, limit?: number | "unbounded"): AsyncGenerator<ParsedHierarchyNode> {
+    let idx = 0;
     for await (const row of queryExecutor.createQueryReader(query, { rowFormat: "ECSqlPropertyNames", ...(limit !== undefined ? { limit } : undefined) })) {
       yield this._props.parser(row);
+      if (idx++ % MAX_ROWS_TO_EMIT_AT_A_TIME === 0) {
```

Comment:
Any reason not to do this in `createLimitingECSqlQueryExecutor`? That way the behavior would be used for other queries other than reading nodes.

Reply:
It was simply much easier to implement it here. But since you think it's worth it, I've moved it to the limiting executor.
Also, since there are multiple for loops and I am not sure which max element count to pick, I've implemented a solution with time measuring (which will probably be more accurate than before)